### PR TITLE
[Term Entry] Go Math Functions Hypot()

### PR DESCRIPTION
**Pull request by MarkTunstall**
### Description

-Added the Hypot() term for Go
-Closes issue #2831 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/go/concepts/math-functions/terms/hypot/hypot.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import math
+
+hypot = math.Sqrt((x * x) + (y * y))
+```
+'Hypot()' accepts two arguments of type 'float64' and returns a value of type 'float64'.
+
+Note:
+- If either argument is negative, the result is positive.
+- If both arguments are negative, the result is positive.
+- If '0' is passed as both arguments, the result is '0'.
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/hypot/hypot.md:30]:
```suggestion
- If `0` is passed as both arguments, the result is `0`. If one of the arguments is `0` the result is equal to the other argument.
```

File: content/go/concepts/math-functions/terms/hypot/hypot.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+```pseudo 
+import math
+
+hypot = math.Sqrt((x * x) + (y * y))
+```
+'Hypot()' accepts two arguments of type 'float64' and returns a value of type 'float64'.
+
+Note:
+- If either argument is negative, the result is positive.
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/hypot/hypot.md:28]:
```suggestion
- If either argument is negative, or both, the result is positive.
```

File: content/go/concepts/math-functions/terms/hypot/hypot.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+- If both arguments are negative, the result is positive.
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/hypot/hypot.md:29]:
```suggestion
- If either argument is positive infinity (`+Inf`), the result will be positive infinity.
```

File: content/go/concepts/math-functions/terms/hypot/hypot.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+- If '0' is passed as both arguments, the result is '0'.
+- If either argument is passed as '0', the result is the other argument.
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/hypot/hypot.md:31]:
```suggestion
- If either argument passed is `NaN`, the result will be `NaN`.
```

File: content/go/concepts/math-functions/terms/hypot/hypot.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+```go
+package main
+
+import("fmt"
+      "math")
+
+func main(){
+    fmt.Printf("Hypot of 3 and 4 = %f\n", math.Sqrt((3 * 3)+(4 * 4)))
+    fmt.Printf("Hypot of -1.5 and -1.5 = %f\n", math.Sqrt((-1.5 * -1.5)+(-1.5 * -1.5)))
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/hypot/hypot.md:44]:
```suggestion
    fmt.Printf("Hypot of 3 and 4 = %f\n", math.Hypot(3 , 4))
    fmt.Printf("Hypot of -1.5 and -1.5 = %f\n", math.Hypot(-1.5,-1.5))
```

File: content/go/concepts/math-functions/terms/hypot/hypot.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+- If either argument is negative, the result is positive.
+- If both arguments are negative, the result is positive.
+- If '0' is passed as both arguments, the result is '0'.
+- If either argument is passed as '0', the result is the other argument.
+
+
+## Example
+
+```go
+package main
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/hypot/hypot.md:37]: The code should be indented with 2 spaces, tabs are not used in docs

File: content/go/concepts/math-functions/terms/hypot/hypot.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+```shell
+Hypot of 3 and 4 = 5.000000
+Hypot of -1.5 and -1.5 = 2.121320
+```
+
+## Codebyte Example (if applicable)
+
+The runnable example of the 'Hypot()' function of the cases where '0' or a negative number is at least one of the arguments. 
+
+```codebyte/golang
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/hypot/hypot.md:59]: make the same corrections here as in the block above

File: content/go/concepts/math-functions/terms/hypot/hypot.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+The above example results in the following output:
+
+```shell
+Hypot of 3 and 4 = 5.000000
+Hypot of -1.5 and -1.5 = 2.121320
+```
+
+## Codebyte Example (if applicable)
+
+The runnable example of the 'Hypot()' function of the cases where '0' or a negative number is at least one of the arguments.
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/hypot/hypot.md:57]:
```suggestion
The runnable example of the `Hypot()` function of the cases where `0` or a negative number is at least one of the arguments. 
```

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@MarkTunstall I've completed my initial review. Please update the entry, and tag me when you've finished making changes. Thanks

File: content/go/concepts/math-functions/terms/hypot/hypot.md [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+```tex
+hypotenuse = \sqrt{(x * x) + (y * y)}
+
+## Syntax
+
+```pseudo 
+import math
+
+result = math.Hypot(x, y)
+```
```
Comment (MarkTunstall) [inline content/go/concepts/math-functions/terms/hypot/hypot.md:27]: result = math.Hypot(x, y)

```

File: content/go/concepts/math-functions/terms/hypot/hypot.md [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+
+The **'Hypot()'** function returns the square root of the sum of two squares.
+
+## Syntax
+
+```pseudo 
+import math
+
+hypot = math.Sqrt((x * x) + (y * y))
+```
```
[the remark this answers is not in the document]
Comment (MarkTunstall) [reply content/go/concepts/math-functions/terms/hypot/hypot.md:26]: thanks @caupolicandiaz, just added spacing in now.

**Comment by MarkTunstall** [comment]
@caupolicandiaz I have commited the changes you suggested.

File: content/go/concepts/math-functions/terms/hypot/hypot.md [hunk 11]
```diff
@@ -0,0 +1,82 @@
+---
+Title: 'Hypot()'
+Description: 'Returns the square root of the sum of two squares.'
+Subjects: 
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags: 
+  - 'Math'
+  - 'Arithmatic'
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/hypot/hypot.md:9]:
```suggestion
  - 'Arithmetic'
```

File: content/go/concepts/math-functions/terms/hypot/hypot.md [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+  - 'Functions'
+CatalogContent: 
+  - 'learn-go'
+  - 'paths/computer-science'
+---
+
+The **`Hypot()`** function returns the square root of the sum of two squares. In other words, the method returns the result of the following formula:
+
+hypotenuse = $`\sqrt{(x * x) + (y * y)}`$
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/hypot/hypot.md:18]: The inline formatting isn't supported in docs. The formula should be inside a code block with the latex call ```tex

File: content/go/concepts/math-functions/terms/hypot/hypot.md [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+Note:
+- If either argument is negative, or both, the result is positive.
+- If either argument is positive infinity (`+Inf`), the result will be positive infinity.
+- If `0` is passed as both arguments, the result is `0`. If one of the arguments is `0` the result is equal to the other argument.
+- If either argument passed is `NaN`, the result will be `NaN`.
+
+
+## Example
+
+```go
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/hypot/hypot.md:39]: The entire code block should be shifted left and the indentations should use 2 spaces, tabs are not used in docs

File: content/go/concepts/math-functions/terms/hypot/hypot.md [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+    fmt.Printf("Hypotenuse of 3 and 4 = %f\n", Hypot(3 , 4))
+    fmt.Printf("Hypotenuse of -1.5 and -1.5 = %f\n", Hypot(-1.5,-1.5))
+    
+  }
+```
+
+The above example results in the following output:
+
+```shell
+  Hypotenuse of 3 and 4 = 5.000000
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/hypot/hypot.md:60]: remove the spaces at the start of these lines

File: content/go/concepts/math-functions/terms/hypot/hypot.md [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+  Hypotenuse of -1.5 and -1.5 = 2.121320
+```
+
+## Codebyte Example 
+
+The runnable example of the `Hypot()` function of the cases where `0` or a negative number is at least one of the arguments. 
+
+```codebyte/golang
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/hypot/hypot.md:68]: same edits as the code block above

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@MarkTunstall I've added a few additional comments. Please update, and tag me when you're finished. Thanks

**Comment by MarkTunstall** [comment]
@caupolicandiaz I've updated the markdown file with your suggestions.

**Review by caupolicandiaz — APPROVED** [review]
@MarkTunstall There were a couple of edits that were not completed accurately. I went ahead and made them directly. Please note that the changes to the example code for future reference. This is ready for a second review.

**Pull request merged**